src/pages/home/sidebar/SidebarLinks.js

```diff
@@ -275,122 +231,5 @@ class SidebarLinks extends React.Component {
-            // eslint-disable-next-line no-param-reassign
-            finalPersonalDetails[accountID] = {
-                accountID: Number(accountID),
-                login: personalData.login,
-                displayName: personalData.displayName,
-                firstName: personalData.firstName,
-                avatar: UserUtils.getAvatar(personalData.avatar, personalData.accountID),
-            };
-            return finalPersonalDetails;
-        },
-        {},
-    );
-
-/**
- * @param {Object} [reportActions]
- * @returns {Object|undefined}
- */
-const reportActionsSelector = (reportActions) =>
-    reportActions &&
-    _.map(reportActions, (reportAction) => ({
-        errors: reportAction.errors,
-        message: [
-            {
-                moderationDecisions: [{decision: lodashGet(reportAction, 'message[0].moderationDecisions[0].decision')}],
-            },
-        ],
-    }));
-
-/**
- * @param {Object} [policy]
- * @returns {Object|undefined}
- */
-const policySelector = (policy) =>
-    policy && {
-        type: policy.type,
-        name: policy.name,
-        avatar: policy.avatar,
-    };
-
-export default compose(
-    withLocalize,
-    withCurrentUserPersonalDetails,
-    withNavigationFocus,
-    withWindowDimensions,
-    withCurrentReportID,
-    withNavigation,
-    withOnyx({
-        // Note: It is very important that the keys subscribed to here are the same
-        // keys that are subscribed to at the top of SidebarUtils.js. If there was a key missing from here and data was updated
-        // for that key, then there would be no re-render and the options wouldn't reflect the new data because SidebarUtils.getOrderedReportIDs() wouldn't be triggered.
-        // This could be changed if each OptionRowLHN used withOnyx() to connect to the Onyx keys, but if you had 10,000 reports
-        // with 10,000 withOnyx() connections, it would have unknown performance implications.
-        chatReports: {
-            key: ONYXKEYS.COLLECTION.REPORT,
-            selector: chatReportSelector,
-        },
-        personalDetails: {
-            key: ONYXKEYS.PERSONAL_DETAILS_LIST,
-            selector: personalDetailsSelector,
-        },
-        priorityMode: {
-            key: ONYXKEYS.NVP_PRIORITY_MODE,
-        },
-        betas: {
-            key: ONYXKEYS.BETAS,
-        },
-        reportActions: {
-            key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
-            selector: reportActionsSelector,
-        },
-        policies: {
-            key: ONYXKEYS.COLLECTION.POLICY,
-            selector: policySelector,
-        },
-        preferredLocale: {
-            key: ONYXKEYS.NVP_PREFERRED_LOCALE,
-        },
-        modal: {
-            key: ONYXKEYS.MODAL,
-        },
-    }),
-)(SidebarLinks);
+export default compose(withLocalize, withCurrentUserPersonalDetails, withWindowDimensions, withNavigation)(SidebarLinks);
```

Comment:
We are currently investigating another issue and it seems that `withNavigation` is no longer used in that component. Or are we missing something?

Reply:
@WoLewicki i think you are right, good catch. There is the usage of `Navigation.js` which uses the global navigation ref. I am not sure if `useNavigation` is there to make sure the navigationRef is ready, but either way i think it can get removed

Reply:
removed it 🥷